docs/advanced.rst

```diff
@@ -1589,24 +1589,56 @@ It is also possible to call python functions via ``operator()``.
     py::object result_py = f(1234, "hello", some_instance);
```

Comment:
Pybind11 has three different ways of passing values between C++ and Python. Custom C++ classes mapped to Python (``class_<>``, convenience wrappers around it like ``stl_binder.h``), Python type wrappers (``py::object, py::array``, etc.), and conversion type casters which convert C++ types into equivalent Python types and vice versa while creating copies in the process (for ``int``, ``std::string``, all stuff in ``stl.h``, ``eigen.h``). I was thinking that it might make sense to split the advanced documentation into three different parts corresponding to each. Thoughts?